src/AoC2024/1/aoc24_1.cpp

```diff
@@ -1,7 +1,66 @@
-#include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <string>
 
-int main(){
-    std::cout << "Hello world!" << '\n';
+#include "aoc24_1.h"
```

Comment:
good practice is to include your files own header first and move (only) the necessary system includes into there.
you basically structure your includes from most specific to most generic. this way most of your files should get the header from the local project from some project header but not the system includes which some of the source files will not need.

Reply:
Agreeing regarding the order.
But wouldn't including the *stream and string libraries in the .cpp file be better with regards to compilation times and encapsulation?

Reply:
yes, exactly it would be. but if your class uses string you'll need the include for it in the header.